src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/SqlDataReader.cs

```diff
@@ -3561,6 +3563,36 @@ private bool TryReadInternal(bool setTimeout, out bool more)
 
                 return true;
             }
+            catch (System.OutOfMemoryException e)
+            {
+                _isClosed = true;
+                SqlConnection con = _connection;
+                if (con != null)
+                {
+                    con.Abort(e);
+                }
+                throw;
+            }
+            catch (System.StackOverflowException e)
+            {
+                _isClosed = true;
+                SqlConnection con = _connection;
+                if (con != null)
+                {
+                    con.Abort(e);
+                }
+                throw;
+            }
+            catch (System.Threading.ThreadAbortException e)
```

Comment:
netcore doesn't have ThreadAbortException anymore. The type exists for compatibility but it will never be thrown. If this code will be merged for the two codebases perhaps leave it in and put a note that it's not needed in netcore